Cargo.toml

```diff
@@ -24,6 +24,7 @@ pomelo = { version = "0.1.4", optional = true }
 thiserror = "1.0.21"
 serde = { version = "1.0", features = ["derive"], optional = true }
 petgraph = { version ="0.5", optional = true }
+rose_tree = { version ="0.2", optional = true }
```

Comment:
is it necessary?  depending on both petgraph and rese_tree seems strange

Reply:
rose_tree is just petgraph's extension that has a concept of a parent. It's directly built on petgraph so it shouldn't really add much. Makes some stuff a little nicer.